e2e_tests/tests/experiment/experiment.py

```diff
@@ -214,74 +199,89 @@ def experiment_has_completed_workload(experiment_id: int) -> bool:
+def experiment_trials_json(experiment_id: int) -> Sequence[bindings.trialv1Trial]:
+    r = bindings.get_GetExperimentTrials(test_session(), experimentId=experiment_id)
+    return r.trials or []
+
+
+def experiment_state(experiment_id: int) -> bindings.determinedexperimentv1State:
+    r = bindings.get_GetExperiment(test_session(), experimentId=experiment_id)
+    return r.experiment.state
+
+
+def experiment_trials(experiment_id: int) -> List[bindings.v1GetTrialResponse]:
+    src_trials = experiment_trials_json(experiment_id)
+    trials = []
+    for trial in src_trials:
+        r = bindings.get_GetTrial(test_session(), trialId=trial.id)
+        trials.append(r)  # includes trial and workload
     return trials
 
 
 def num_experiments() -> int:
     certs.cli_cert = certs.default_load(conf.make_master_url())
     authentication.cli_auth = authentication.Authentication(conf.make_master_url(), try_reauth=True)
     r = api.get(conf.make_master_url(), "experiments")
-    assert r.status_code == requests.codes.ok, r.text
     return len(r.json())
 
 
 def cancel_single(experiment_id: int, should_have_trial: bool = False) -> None:
     cancel_experiment(experiment_id)
 
-    trials = experiment_trials(experiment_id)
-    if should_have_trial or len(trials) > 0:
-        assert len(trials) == 1, len(trials)
-
-        trial = trials[0]
-        assert trial["state"] == "CANCELED", trial["state"]
-
-
-def cancel_single_v1(experiment_id: int, should_have_trial: bool = False) -> None:
-    cancel_experiment_v1(experiment_id)
-
-    trials = experiment_trials(experiment_id)
-    if should_have_trial or len(trials) > 0:
+    if should_have_trial:
+        trials = experiment_trials(experiment_id)
         assert len(trials) == 1, len(trials)
 
-        trial = trials[0]
-        assert trial["state"] == "CANCELED", trial["state"]
+        trial = trials[0].trial
+        assert trial.state == bindings.determinedexperimentv1State.STATE_CANCELED
 
 
 def is_terminal_state(state: str) -> bool:
-    return state in ("CANCELED", "COMPLETED", "ERROR")
+    return state in (
+        bindings.determinedexperimentv1State.STATE_CANCELED.value,
+        bindings.determinedexperimentv1State.STATE_COMPLETED.value,
+        bindings.determinedexperimentv1State.STATE_ERROR.value,
+    )
 
 
 def trial_metrics(trial_id: int) -> Dict[str, Any]:
     certs.cli_cert = certs.default_load(conf.make_master_url())
     authentication.cli_auth = authentication.Authentication(conf.make_master_url(), try_reauth=True)
     r = api.get(conf.make_master_url(), "trials/{}/metrics".format(trial_id))
-    assert r.status_code == requests.codes.ok, r.text
     json = r.json()  # type: Dict[str, Any]
     return json
 
 
 def get_flat_metrics(trial_id: int, metric: str) -> List:
     full_trial_metrics = trial_metrics(trial_id)
-    metrics = [m for step in full_trial_metrics["steps"] for m in step["metrics"]["batch_metrics"]]
+    metrics = [
```

Comment:
this doesn't looks right to me, since `trial_metrics` should still be returning steps. but it looks like `get_flat_metrics` is unused so it doesn't matter?

Reply:
this is no longer used so I will remove it